File: 백준/Silver/18258. 큐 2/큐 2.js
const [n, ...input] = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

class Queue {
    constructor() {
        this.data = [];
        this.head = 0;
        this.tail = 0;
    }
    push(data) {
        this.data[this.tail++] = data;
    }
    pop() {
        if (this.head === this.tail) return '-1';
        let p = this.data[this.head];
        this.head++;
        return p;
    }
    size() {
        return this.tail - this.head;
    }
    empty() {
        return this.tail - this.head === 0 ? '1' : '0';
    }
    front() {
        if (this.tail === this.head) return '-1';
        return this.data[this.head];
    }
    back() {
        if (this.tail === this.head) return '-1';
        return this.data[this.tail - 1];
    }
}

const queue = new Queue();
const answer = [];
for (let i = 0; i < +n; i++) {
    const [cmd, number] = input[i].split(' ');
    if (cmd === 'push') queue.push(number);
    else if (cmd === 'pop') answer.push(queue.pop());
    else if (cmd === 'size') answer.push(queue.size());
    else if (cmd === 'empty') answer.push(queue.empty());
    else if (cmd === 'front') answer.push(queue.front());
    else answer.push(queue.back());
}
console.log(answer.join('\n'));

File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <iostream>
#include <queue>
using namespace std;

int t, s, e, i;
int q[100000000];

int main(void){
  ios_base::sync_with_stdio(0);
  cin.tie(0);  
  cin >> t;
  while(t--){
    string cmd; 
    cin >> cmd;
    if(cmd == "push"){
      cin >> q[e++];
    }else if(cmd == "pop"){
      cout << (s == e? -1 : q[s++]) << '\n';
    }else if(cmd == "size"){
      cout << e - s << '\n';
    }else if(cmd == "empty"){
      cout << (s == e) << '\n';
    }else if(cmd == "front"){
      cout << (s == e? -1 : q[s]) << '\n'; 
    }else{
      cout << (s == e? -1 : q[e-1]) << '\n';
    }
  }
}